File: examples/key/new_local_certificate/new_local_certificate.py
from bloock.client.key import KeyClient
from bloock.entity.key.key_type import KeyType
from bloock.entity.key.local_certificate_params import LocalCertificateParams
from bloock.entity.key.subject_certificate_params import SubjectCertificateParams

if __name__ == "__main__":
    key_client = KeyClient()

    key_type = KeyType.Rsa2048
    subject_params = SubjectCertificateParams(
        "Common name",
        "Organization",
        "Organization Unit",
        "Location",
        "State",
        "US"
    )
    params = LocalCertificateParams(key_type, subject_params, "password", 2)
    local_certificate = key_client.new_local_certificate(params)

File: examples/key/new_local_certificate/new_local_certificate.php
<?php

require "./vendor/autoload.php";

use Bloock\Client\KeyClient;
use Bloock\Entity\Key\KeyType;
use Bloock\Entity\Key\SubjectCertificateParams;
use Bloock\Entity\Key\LocalCertificateArgs;

$keyClient = new KeyClient();

$keyType = KeyType::Bjj;
$subjectParams = new SubjectCertificateParams(
  "Common Name",
  "Organization",
  "Organization Unit",
  "Location",
  "State",
  "US"
);

$params = new LocalCertificateArgs($keyType, $subjectParams, "password", 2);
$certificate = $keyClient->newLocalCertificate($params);
